src/libraries/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoder.cs

```diff
@@ -29,7 +29,12 @@ public abstract class TextEncoder
         // Fast cache for Ascii
         private readonly byte[][] _asciiEscape = new byte[0x80][];
 
-        private readonly int[] _asciiNeedsEscaping = new int[0x80];
+        private bool _isAsciiCacheInitialized;
+        private readonly bool[] _asciiNeedsEscaping = new bool[0x80];
```

Comment:
Changed to `bool[]` as I didn't see any benefit from using `int[]` with `1` or `-1` (and forgot to ask in the review in that PR).